insonmnia/dwh/server.go

```diff
@@ -387,24 +542,41 @@ func (w *DWH) getProfileInfo(ctx context.Context, request *pb.ID) (*pb.Profile,
 func (w *DWH) getProfileInfoTx(tx *sql.Tx, request *pb.ID) (*pb.Profile, error) {
 	rows, err := tx.Query(w.commands["selectProfileByID"], request.Id)
 	if err != nil {
-		return nil, err
+		w.logger.Error("failed to selectProfileByID", zap.Error(err), zap.Any("request", request))
+		return nil, status.Error(codes.Internal, "failed to GetProfileInfo")
 	}
 	defer rows.Close()
 
 	if ok := rows.Next(); !ok {
-		return nil, errors.Errorf("profile `%s` not found", request.Id)
+		w.logger.Error("profile not found", zap.Error(err), zap.Any("request", request))
+		return nil, status.Error(codes.Internal, "failed to GetProfileInfo")
 	}
 
 	return w.decodeProfile(rows)
 }
 
-func (w *DWH) GetBlacklist(ctx context.Context, adderID *pb.ID) (*pb.BlacklistReply, error) {
+func (w *DWH) GetBlacklist(ctx context.Context, request *pb.BlacklistRequest) (*pb.BlacklistReply, error) {
 	w.mu.RLock()
 	defer w.mu.RUnlock()
 
-	rows, err := w.db.Query(w.commands["selectBlacklists"], adderID.Id)
+	return w.getBlacklist(ctx, request)
+}
+
+func (w *DWH) getBlacklist(ctx context.Context, request *pb.BlacklistRequest) (*pb.BlacklistReply, error) {
+	var filters []*filter
+	if request.OwnerID > "0" {
```

Comment:
why "0" ? What if ownerID is an empty string? Maybe try to typecast into `common.address`?

Reply:
If `OwnerID` is an empty string, it will be less than `"0"`.